test/framework/src/main/java/org/elasticsearch/index/mapper/MapperScriptTestCase.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.script.ScriptContext;
+
+import java.io.IOException;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.Matchers.arrayWithSize;
+import static org.hamcrest.Matchers.equalTo;
+
+public abstract class MapperScriptTestCase<FactoryType> extends MapperServiceTestCase {
+
+    protected abstract String type();
+
+    protected abstract FactoryType serializableScript();
+
+    protected abstract FactoryType errorThrowingScript();
+
+    protected FactoryType compileScript(String name) {
+        throw new UnsupportedOperationException("Unknown script " + name);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected final <T> T compileScript(Script script, ScriptContext<T> context) {
+        if (script.getIdOrCode().equals("serializer_test")) {
+            return (T) serializableScript();
+        }
+        if (script.getIdOrCode().equals("throws")) {
+            return (T) errorThrowingScript();
+        }
+        return (T) compileScript(script.getIdOrCode());
+    }
+
+    public void testSerialization() throws IOException {
```

Comment:
nit: call it testToXContent? I somehow think of serialization as the transport one, so I get confused with this naming.